parse/functions.go

```diff
@@ -175,6 +175,90 @@ var (
 				return fmt.Sprintf("uuid_generate_v5(%s)", strings.Join(inputs, ", ")), nil
 			},
 		},
+		"encode": {
+			ValidateArgs: func(args []*types.DataType) (*types.DataType, error) {
+				// first must be blob, second must be text
+				if len(args) != 2 {
+					return nil, wrapErrArgumentNumber(2, len(args))
+				}
+
+				if !args[0].EqualsStrict(types.BlobType) {
+					return nil, wrapErrArgumentType(types.BlobType, args[0])
+				}
+
+				if !args[1].EqualsStrict(types.TextType) {
+					return nil, wrapErrArgumentType(types.TextType, args[1])
+				}
+
+				return types.TextType, nil
+			},
+			PGFormat: func(inputs []string, distinct bool, star bool) (string, error) {
+				if star {
+					return "", errStar("encode")
+				}
+				if distinct {
+					return "", errDistinct("encode")
+				}
+
+				return fmt.Sprintf("encode(%s)", strings.Join(inputs, ", ")), nil
+			},
+		},
+		"decode": {
+			ValidateArgs: func(args []*types.DataType) (*types.DataType, error) {
```

Comment:
Do we have function return validation? 
When we validate input for example of a foregin procedure call, we will need it right?

Reply:
ah I see, it's already defined in the `ValidateArgs`, all good.